packages/website/src/components/Playground.tsx

```diff
@@ -168,57 +155,56 @@ function Playground(): React.JSX.Element {
             <EditorTabs
               tabs={detailTabs}
               active={state.showAST ?? false}
-              change={(v): void => setState({ showAST: v })}
+              change={showAST => setState({ showAST })}
             />
             {state.showAST === 'es' && (
               <ESQueryFilter
                 defaultValue={state.esQuery?.filter}
                 onChange={(filter, selector) =>
-                  setState({
-                    esQuery: { filter, selector },
-                  })
+                  setState({ esQuery: { filter, selector } })
                 }
                 onError={setEsQueryError}
               />
             )}
           </div>
           <div className={styles.playgroundInfoContainer}>
-            {(state.showAST === 'es' && esQueryError && (
+            {state.showAST === 'es' && esQueryError ? (
               <ErrorViewer
                 type="warning"
                 title="Invalid Selector"
                 value={esQueryError}
               />
-            )) ||
-              (state.showAST === 'types' && astModel?.storedTsAST && (
+            ) : state.showAST && astModel ? (
+              state.showAST === 'types' ? (
                 <TypesDetails
                   typeChecker={astModel.typeChecker}
-                  value={astModel.storedTsAST}
+                  value={
+                    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                    astModel.storedTsAST!
+                  }
                   onHoverNode={setSelectedRange}
                   cursorPosition={position}
                 />
-              )) ||
-              (state.showAST && astModel && (
+              ) : (
                 <ASTViewer
```

Comment:
that looks ok to me.

-----

i was meaning to rewrite this at some point to use provider, passing events back and forward make all this code so unreadable and requires us to use stuff like this